File: bike-shop/backend/src/main/java/com/example/bike_shop/service/FavoriteService.java
package com.example.bike_shop.service;

import com.example.bike_shop.dto.FavoriteDTO;

import java.util.List;

public interface FavoriteService {
    List<FavoriteDTO> getByUser(Long userId);
    FavoriteDTO add(Long userId, Long productId);
    void remove(Long userId, Long productId);
    boolean isFavorite(Long userId, Long productId);
}

File: bike-shop/frontend/src/services/favoriteService.js
import axiosClient from "./axiosConfig";

const favoriteService = {
  getAll: () => axiosClient.get("/favorite"),
  isFavorite: (productId) => axiosClient.get(`/favorite/${productId}`),
  add: (productId) => axiosClient.post(`/favorite/${productId}`),
  remove: (productId) => axiosClient.delete(`/favorite/${productId}`),
};

export default favoriteService;
